src/compiler/scanner.ts

```diff
@@ -998,6 +1033,25 @@ module ts {
                     return token = SyntaxKind.EndOfFileToken;
                 }
                 let ch = text.charCodeAt(pos);
+
+                if (inJSXChild) {
+                    switch (ch) {
+                        case CharacterCodes.closeBrace:

```

Comment:
I guess I don't see how these two cases are different from how we treat closebrace/openbrace in the switch below.